src/internal/pfsdb/repos.go

```diff
@@ -0,0 +1,388 @@
+package pfsdb
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/coredb"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+const (
+	getRepoAndBranches = "SELECT repo.id, repo.name, repo.type, repo.project_id, " +
+		"repo.description, array_agg(branch.proto) AS branches, repo.created_at, repo.updated_at FROM pfs.repos repo " +
+		"LEFT JOIN core.projects project ON repo.project_id = project.id " +
```

Comment:
I think we can safely just use a `JOIN` on projects because a repo is guaranteed to belong to a project.